integration-tests/runtime-tests/test/tests/mosaic/mosaicTests.ts

```diff
@@ -0,0 +1,419 @@
+                this.skip();
+            const toTransfer = userWallet.address;
+            await TxMosaicTests.lockFunds(startRelayerWallet, 
+                pNetworkId, 
+                remoteAssetId, 
+                toTransfer, 
+                transferAmount);
+            const lockedAmount = await api.query.mosaic.incomingTransactions(toTransfer, assetId);
+            //verify that the incoming transaction is locked in the incoming transaction pool.
+            expect(lockedAmount.unwrap()[0].toNumber()).to.be.equal(transferAmount);
+        });
+
+        it('Other users should be able to mint assets into pallet wallet with timelock//simulates incoming transfer from pair network', async function(){
+            // Check if this test is enabled.
+            if (!testConfiguration.enabledTests.userCanMintAssets)
+                this.skip();
+            const toTransfer = newRelayerWallet.address;
+            const pAmount = api.createType('u128', transferAmount);
+            await TxMosaicTests.lockFunds(startRelayerWallet, 
+                pNetworkId, 
+                remoteAssetId, 
+                toTransfer, 
+                pAmount);
+            const lockedAmount = await api.query.mosaic.incomingTransactions(toTransfer, assetId);
+            //verify that the incoming transaction is locked in the incoming transaction pool.
+            expect(lockedAmount.unwrap()[0].toNumber()).to.be.equal(transferAmount);
+        });
+        it('Only relayer should mint assets into pallet wallet with timelock/incoming transactions', async function(){
+            // Check if this test is enabled.
+            if (!testConfiguration.enabledTests.OnlyRelayerCanMintAssets)
+                this.skip();
+            const toTransfer = newRelayerWallet.address;
+            const pAmount = api.createType('u128', transferAmount);
+            //verify that the transaction fails with BadOrigin message
+            await TxMosaicTests.lockFunds(userWallet, 
+                pNetworkId, 
+                remoteAssetId, 
+                toTransfer, 
+                pAmount).catch(error=>
+                expect(error.message).to.contain("BadOrigin"));
+        });
+        /**
+         * Rotating the relayer.
+         * Sudo call therefore result is checked by `.isOk`.
+         */
+        it('Should be able to rotate relayer', async function () {
+            if (!testConfiguration.enabledTests.query.account__success.balanceGTZero1)
+               this.skip();
+            const {data:[result]} = await TxMosaicTests.testRotateRelayer(startRelayerWallet, 
+                newRelayerWallet.address);
+            const relayerInfo = await api.query.mosaic.relayer();
+            //verify that the relayer records information about the next relayer wallet
+            expect(relayerInfo.unwrap().relayer.next.toJSON().account).to.be.equal(api.createType('AccountId32', newRelayerWallet.address).toString());
```

Comment:
This only compiles if the code is compiled without type checking cause the linter seems to block this. 
I couldn't get it quickly fixed. We can try on this together tomorrow.

```
> tsc -p . --declaration

test/tests/mosaic/mosaicTests.ts:200:41 - error TS2339: Property 'relayer' does not exist on type 'PalletMosaicRelayerStaleRelayer'.

200             expect(relayerInfo.unwrap().relayer.next.toJSON().account).to.be.equal(api.createType('AccountId32', newRelayerWallet.address).toString());
                                            ~~~~~~~


Found 1 error in test/tests/mosaic/mosaicTests.ts:200
```